validate-differential-expression/gene_DE_validate.Rmd

```diff
@@ -0,0 +1,471 @@
+include a metadata in a separate TSV file. 
+This data set was obtained from [refine.bio](https://www.refine.bio/). 
+
+```{r}
+# Put name of data file
+refine.bio.data.filename <- file.path("data", "GSE37418.tsv")
+
+# Unzip the file and direct it to the data folder
+unzip(paste0(refine.bio.data.filename, ".zip"), exdir = "data")
+
+# Read in data tsv file
+refine.bio.df <- readr::read_tsv(refine.bio.data.filename, progress = FALSE) %>% 
+  as.data.frame() %>%
+  # Make the gene column the rownames so the gene names are out of the way for calculations
+  tibble::column_to_rownames('X1') 
+```
+
+Set up the metadata for this set.
+
+```{r}
+# Metadata file name
+metadata.filename <- file.path("data", "metadata_GSE37418.tsv")
+
+# Read in metadata tsv file
+metadata <- readr::read_tsv(metadata.filename) %>% 
+  # Get rid of columns that have only NAs
+  dplyr::select(-which(apply(is.na(.), 2, all)))
+
+# Print out metadata so we can get an idea of what we have
+metadata
+```
+
+Let's reformat our variable `subgroup` that we will be using for testing. 
+To make the comparisons more reasonably similar, we will get rid of samples 
+that *aren't* SHH, Group3, or Group4.
+
+```{r}
+# Reformat subgroup names to make Group 3 and 4 into "NonSHH"
+metadata$subgroup <- gsub("G3|G4", "NonSHH", metadata$subgroup)
+
+# Check out subgroup summary
+summary(as.factor(metadata$subgroup))
+```
+
+To make this dataset more comparable to our imported dataset (which only has SHH,
+Group3, or Group4), we will get rid of samples that *aren't* in one of these 
+classifications.
+
+```{r}
+# Only keep samples that are SHH or Non-SHH (which is now both Group3 and Group4)
+metadata <- metadata %>% 
+  dplyr::filter(subgroup %in% c("SHH", "NonSHH"))
+```
+
+Make refine.bio metadata and data are same samples, in the same order.
+
+```{r}
+# Reorder/filter data to match metadata
+refine.bio.df <- refine.bio.df[, match(metadata$refinebio_accession_code, 
+                                       colnames(refine.bio.df))]
+
+# Check if they are the same:
+all.equal(colnames(refine.bio.df), metadata$refinebio_accession_code)
+```
+
+## 5) Perform differential expression analysis on refine.bio data
+Here we are using the same model and set up as we used for the Northcott data
+in [`author_processed_DE`](./author_processed_DE.nb.html). 
+`limma` needs a numeric design matrix to signify which samples belong to which 
+groups you are testing. 
+Note that we are using the medulloblastoma subgroups as our variable for testing.
+This is the same type of variable we used for our imported author processed 
+dataset (See [`author_processed_DE`](./author_processed_DE.nb.html)).
+
+```{r Set up design matrix}
+# Make subgroup a factor
+metadata$subgroup <- as.factor(metadata$subgroup)
+
+# Create the design matrix
+des.mat <- model.matrix(~ metadata$subgroup)
+
+# Make the column name more informative
+colnames(des.mat)[2] <- paste0(levels(subgroup), collapse = "vs")
```

Comment:
This is broken for me currently. I believe it should be:

```suggestion
colnames(des.mat)[2] <- paste0(levels(metadata$subgroup), collapse = "vs")
```

Were you able to run this notebook in its entirely after restarting R?